### ISSUE TITLE
`consistent-chaining` false positives

### ISSUE THREAD

**Issue by MrVauxs** [antfu/eslint-plugin-antfu#31]
### Describe the bug

The bug pertains to the new `consistent-chaining` rule. As shown in the documentation, the following should be correct
```js
const foo2 = []
  .map(x => x + 'bar')
  .filter(Boolean)
  ```
However, the following code returns an error
```js
const x = {foo: 1, bar: 2 }
const result = Object.keys(x)
  .reduce((acc, key) => ({ ...acc, ...x[key] }), {});
```

![Code_dOtkx0AXeO](https://github.com/user-attachments/assets/79f90531-4636-4700-8843-ec7179351b9d)

It looks like the rule explicitly expects the first property in the chain to be a variable. If its a function, it breaks.

![Code_DGaP0eaHUt](https://github.com/user-attachments/assets/9045580d-a400-4c93-b2c8-d4bb5ff9369e)

### Reproduction

https://github.com/MrVauxs/antfu-repro

### System Info

```Shell
System:
    OS: Windows 11 10.0.22631
    CPU: (12) x64 AMD Ryzen 5 3600 6-Core Processor
    Memory: 12.42 GB / 31.93 GB
  Binaries:
    Node: 22.6.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.8.3 - C:\Program Files\nodejs\npm.CMD
    bun: 1.1.12 - ~\.bun\bin\bun.EXE
  Browsers:
    Edge: Chromium (127.0.2651.74)
    Internet Explorer: 11.0.22621.3527
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### Contributions

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests (actually just go ahead and do it, thanks!)

**Comment by subframe7536** [comment]
Also, properties on `this` should also not be wrapped

```ts
const result = this.userList
  .filter(user => user.name.startWith('T'))
  .map(user => user.phoneNumber)
```

**Comment by antfu** [comment]
https://github.com/antfu/eslint-plugin-antfu/commit/10e9920e51d691635016921dc72700e2624726c0 should fix it. thanks
